pkgs/development/interpreters/python/hooks/fix-python-deps.sh

```diff
@@ -0,0 +1,76 @@
+# shellcheck shell=bash
+
+# Setup hook that fixes Python dependencies versions.
+#
+# Example usage in a derivation:
+#
+#   { …, pythonPackages, … }:
+#
+#   pythonPackages.buildPythonPackage {
+#     …
+#     nativeBuildInputs = [ pythonPackages.fixPythonDepsHook ];
+#
+#     # Already the default, only change if necessary
+#     requirementsTxt = "requirements.txt";
+#
+#     # This will remove the upperbound version of the package
+#     # e.g.: abc>1<=2 -> abc>1
+#     relaxPythonDeps = [ "abc" ];
+#     # This will remove all dependencies restrictions
+#     # e.g.: cde>1<=2 -> cde
+#     unlockPythonDeps = [ "cde" ];
+#     # This will remove the dependency
+#     # e.g.: fgh>1<=2 -> <nothing>
+#     removePythonDeps = [ "cde" ];
+#     …
+#   }
+#
+# This hook will copy files which are either given by full path
+# or all '*.desktop' files placed inside the 'share/applications'
+# folder of each `desktopItems` argument.
+
+declare -a relaxPythonDeps
+declare -a unlockPythonDeps
+declare -a removePythonDeps
+
+requirementsTxt="${requirementsTxt:-requirements.txt}"
```

Comment:
Can we search for a file that matches ``requirements*.txt``?

Reply:
Huh... It can actually, however the way I wrote this it will probably fail if this glob matches more than one file.

I think this can be fixed, though. And allowing arbitrary globs is a good thing so people can do more crazy things (like `requirements{,_dev}.txt`).

Reply:
Done.

Reply:
Still thinking if it would be a good idea to also accept a vector for example, so we could have multiple files being passed.

But maybe that is too much.